client/kv.go

```diff
@@ -78,12 +70,28 @@ type KV struct {
 // create a transactional client, the KV struct should be manually
 // initialized in order to utilize a txnSender. Clock is used to
 // formulate client command IDs, which provide idempotency on API
-// calls. If clock is nil, uses time.UnixNanos as default
+// calls and defaults to the system clock.
 // implementation.
-func NewKV(sender KVSender, clock Clock) *KV {
+func NewKV(ctx *Context, sender KVSender) *KV {
+	if ctx == nil {
+		ctx = NewContext()
+	}
 	return &KV{
-		sender: sender,
-		clock:  clock,
+		sender:          sender,
+		User:            ctx.User,
```

Comment:
And export it..then you can just set the db.Context.TxnRetryOptions manually in txn_correctness_test and not directly access the client.DefaultTxnRetryOptions, which is ugly. Is there some reason this wouldn't work?